### PR TITLE
Hyperdrive notebook updates

### PR DESCRIPTION
**Pull request by PatrickBue**
### Description
Updated hyperdrive notebook and added pytest

### CODE REVIEW

**Comment by PatrickBue** [comment]
The newly added AML notebook test is passing, as can be seen in this manually started test run:
https://dev.azure.com/best-practices/computervision/_build/results?buildId=10527&view=ms.vss-test-web.build-test-results-tab

File: scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb [hunk 1]
```diff
@@ -41,14 +41,17 @@
   {
```
Comment (jiata) [inline scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>this image doesn't reflect what new 'data sub folder'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/ComputerVision/pull/356/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (jiata) [inline scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>can we delete these commented out lines:</p><p> </p><p># Get accuracy on test set at IOU=0.5</p><p>#e = detector.evaluate()</p><p>#coco_eval = e.coco_eval["bbox"].eval["precision"]</p><p>#precision_settings = (0, slice(0, None), slice(0, None), 0, 2)</p><p>#acc = np.mean(np.mean(coco_eval[precision_settings]))</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/ComputerVision/pull/356/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (jiata) [inline scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>[minor] can we run the black formatter on this cell?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/ComputerVision/pull/356/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (jiata) [inline scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>[minor] we should stick to the convention of this formatting;</p><p><code></code><code></p><p>var_assignment = a_really_long_function(</p><p>  param_1,</p><p>  param_2,</p><p>  param_3</p><p>)</p><p></code><code></code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/ComputerVision/pull/356/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

**Comment by jiata** [comment]
addresses #337

File: scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb (hunk 1 above)
Comment (PatrickBue) [reply scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: Can you clarify what you mean? The content that's shown (the annotation and images directories) is inside the data_subfolder

Comment (PatrickBue) [reply scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: deleted

Comment (PatrickBue) [reply scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>Done. Looks a lot nicer now</p>

Comment (PatrickBue) [reply scenarios/detection/11_exploring_hyperparameters_on_azureml.ipynb]: <p>Agreed. I also ran most of the other cells through black formatter to get better formatting. That said, there was one cell in particular where the current version is much nicer to the eye.</p>

**Pull request merged**